src/main/java/org/opensearch/ad/ExecuteADResultResponseRecorder.java

```diff
@@ -0,0 +1,278 @@
+                        null,
+                        r.getTotalUpdates(),
+                        response.getDetectorIntervalInMinutes(),
+                        response.getError()
+                    );
+                }, e -> { log.error("Failed to update latest realtime task for " + detectorId, e); }));
+            };
+            if (!adTaskManager.isHCRealtimeTaskStartInitializing(detectorId)) {
+                // real time init progress is 0 may mean this is a newly started detector
+                // Delay real time cache update by one minute. If we are in init status, the delay may give the model training time to
+                // finish. We can change the detector running immediately instead of waiting for the next interval.
+                threadPool.schedule(profileHCInitProgress, new TimeValue(60, TimeUnit.SECONDS), AnomalyDetectorPlugin.AD_THREAD_POOL_NAME);
+            } else {
+                profileHCInitProgress.run();
+            }
+
+        } else {
+            log
+                .debug(
+                    "Update latest realtime task for single stream detector {}, total updates: {}",
+                    detectorId,
+                    response.getRcfTotalUpdates()
+                );
+            updateLatestRealtimeTask(
+                detectorId,
+                null,
+                response.getRcfTotalUpdates(),
+                response.getDetectorIntervalInMinutes(),
+                response.getError()
+            );
+        }
+    }
+
+    private void updateLatestRealtimeTask(
+        String detectorId,
+        String taskState,
+        Long rcfTotalUpdates,
+        Long detectorIntervalInMinutes,
+        String error
+    ) {
+        // Don't need info as this will be printed repeatedly in each interval
+        adTaskManager
+            .updateLatestRealtimeTaskOnCoordinatingNode(
+                detectorId,
+                taskState,
+                rcfTotalUpdates,
+                detectorIntervalInMinutes,
+                error,
+                ActionListener.wrap(r -> {
+                    if (r != null) {
+                        log.debug("Updated latest realtime task successfully for detector {}, taskState: {}", detectorId, taskState);
+                    }
+                }, e -> {
+                    if ((e instanceof ResourceNotFoundException) && e.getMessage().contains(CAN_NOT_FIND_LATEST_TASK)) {
+                        // Clear realtime task cache, will recreate AD task in next run, check AnomalyResultTransportAction.
+                        log.error("Can't find latest realtime task of detector " + detectorId);
+                        adTaskManager.removeRealtimeTaskCache(detectorId);
+                    } else {
+                        log.error("Failed to update latest realtime task for detector " + detectorId, e);
+                    }
+                })
+            );
+    }
+
+    public void indexAnomalyResultException(
```

Comment:
can you add a comment here that it's not only indexing the result with the exception, but also updating the task state after 60s if the exception is related to cold start (index not found exceptions)?

Reply:
added